src/scripting/lua_ui.cc

```diff
@@ -920,513 +1120,1270 @@ static std::function<void()> create_plugin_action_lambda(lua_State* L, const std
+	if (reg.window != nullptr) {
+		return reg.window;
+	}
+
+	std::string name = get_table_string(L, "name", true);
+	std::string title = get_table_string(L, "title", true);
+
+	std::string tooltip = get_table_string(L, "tooltip", false);
+	int32_t x = get_table_int(L, "x", false);
+	int32_t y = get_table_int(L, "y", false);
+	int32_t w = get_table_int(L, "w", false);
+	int32_t h = get_table_int(L, "h", false);
+
+	UI::UniqueWindow* window =
+	   new UI::UniqueWindow(parent, UI::WindowStyle::kWui, name, &reg, x, y, w, h, title);
+
+	lua_getfield(L, -1, "content");
+	if (!lua_isnil(L, -1)) {
+		window->set_center_panel(do_create_child(L, window, nullptr));
 	}
+	lua_pop(L, 1);
+
+	return window;
+}
+
+UI::Panel* LuaPanel::do_create_child_window(lua_State* L, UI::Panel* parent) {
+	if (parent != get_egbase(L).get_ibase()) {
+		report_error(L, "Windows must be toplevel components");
+	}
+
+	std::string tooltip = get_table_string(L, "tooltip", false);
+	int32_t x = get_table_int(L, "x", false);
+	int32_t y = get_table_int(L, "y", false);
+	int32_t w = get_table_int(L, "w", false);
+	int32_t h = get_table_int(L, "h", false);
+
+	std::string name = get_table_string(L, "name", true);
+	std::string title = get_table_string(L, "title", true);
+	UI::Window* window = new UI::Window(parent, UI::WindowStyle::kWui, name, x, y, w, h, title);
+
+	lua_getfield(L, -1, "content");
+	if (!lua_isnil(L, -1)) {
+		window->set_center_panel(do_create_child(L, window, nullptr));
+	}
+	lua_pop(L, 1);
+
+	return window;
+}
+
+/* RST
+Button
+------
+
+.. class:: Button
+
+   This represents a simple push button.
+*/
+const char LuaButton::className[] = "Button";
+const MethodType<LuaButton> LuaButton::Methods[] = {
+   METHOD(LuaButton, press),         METHOD(LuaButton, click),
+   METHOD(LuaButton, set_repeating), METHOD(LuaButton, set_perm_pressed),
+   METHOD(LuaButton, toggle),        {nullptr, nullptr},
+};
+const PropertyType<LuaButton> LuaButton::Properties[] = {
+   PROP_RW(LuaButton, title),
+   PROP_RW(LuaButton, enabled),
+   {nullptr, nullptr, nullptr},
+};
+
+/*
+ * Properties
+ */
+
+/* RST
+   .. attribute:: title
+
+      .. versionadded:: 1.2
+
+      (RW) The text shown on the button.
+*/
+int LuaButton::get_title(lua_State* L) {
+	lua_pushstring(L, get()->get_title().c_str());
+	return 1;
+}
+int LuaButton::set_title(lua_State* L) {
+	get()->set_title(luaL_checkstring(L, -1));
+	return 0;
+}
+
+/* RST
+   .. attribute:: enabled
+
+      .. versionadded:: 1.2
+
+      (RW) Whether the user may interact with the button.
+*/
+int LuaButton::get_enabled(lua_State* L) {
+	lua_pushboolean(L, static_cast<int>(get()->enabled()));
+	return 1;
+}
+int LuaButton::set_enabled(lua_State* L) {
+	get()->set_enabled(luaL_checkboolean(L, -1));
+	return 0;
+}
+
+/*
+ * Lua Functions
+ */
+/* RST
+   .. method:: press()
+
+      Press and hold this button. This is mainly to visualize a pressing
+      event in tutorials
+*/
+int LuaButton::press(lua_State* /* L */) {
```

Comment:
Is there also a `release()`? Or is this achieved by calling `click()`? But this could lead to unintended double activation.

Reply:
That's done by calling `click()`. Just calling `press()` does not activate the button. I clarified this in the doc.

A lot of these legacy functions have been around for ages for very specialized use cases and don't really fit to modern design principles, but have to be retained anyway for compatibility.